src/utils/derivation-utils.ts

```diff
@@ -1,44 +1,191 @@
+    }
+
+    return {
+        name: tableName,
+        source,
+        lambda: evaluatedLambda,
+        shuffle: shuffle ?? 'any',
+    };
+};
+
+const generateSqlTemplate = (
+    entityName: string,
+    sourceCollections: Set<string> | string[],
+    options?: TemplateOptions
+) => {
+    const existingTransforms = options?.existingTransforms;
+
+    const existingMigrations = options?.existingMigrations;
+    const templateFiles = options?.templateFiles;
+
+    let transforms: Transform[] = [];
+
+    if (existingTransforms && existingTransforms.length > 0) {
+        transforms = existingTransforms.map(({ collection, lambda, shuffle }) =>
+            formatSqlTransforms(
+                collection,
+                entityName,
+                templateFiles,
+                lambda,
+                shuffle
+            )
+        );
+    } else if (isArray(sourceCollections)) {
+        transforms = sourceCollections.map((source) =>
+            formatSqlTransforms(source, entityName, templateFiles)
+        );
+    } else {
+        transforms = Array.from(sourceCollections).map((source) =>
+            formatSqlTransforms(source, entityName, templateFiles)
+        );
+    }
+
+    let sqlite: { migrations?: string[] } = {};
+
+    if (existingMigrations && existingMigrations.length > 0) {
+        sqlite = { migrations: existingMigrations };
+    } else if (templateFiles) {
+        sqlite = { migrations: [`${entityName}.migration.0.sql`] };
+    }
+
+    return {
+        schema,
+        key,
+        derive: {
+            // TODO (GitPod) what do we make this base here?
+            //  there could be multiple bases
+            using: { sqlite },
+            transforms,
+        },
+    };
+};
+
+const formatTsTransforms = (source: string) => {
+    const baseName = stripPathing(source);
+
+    return {
+        name: `${baseName}`,
+        shuffle: 'any',
+        source,
+    };
+};
+
+const generateTsTemplate = (
+    entityName: string,
+    sourceCollections: Set<string> | string[]
+) => {
+    const transforms = isArray(sourceCollections)
+        ? sourceCollections.map(formatTsTransforms)
+        : Array.from(sourceCollections).map(formatTsTransforms);
+
+    return {
+        schema,
+        key,
+        derive: {
+            using: { typescript: { module: `${entityName}.ts` } },
+            transforms,
+        },
+    };
+};
+
+export const generateInitialSpec = (
+    language: DerivationLanguage,
+    entityName: string,
+    sourceCollections: Set<string> | string[],
+    options?: TemplateOptions
+) => {
+    const strippedEntityName = stripPathing(entityName);
+
+    switch (language) {
+        case 'typescript':
+            return generateTsTemplate(strippedEntityName, sourceCollections);
+        default:
+            return generateSqlTemplate(
+                strippedEntityName,
+                sourceCollections,
+                options
+            );
```

Comment:
No change needed: I think as we get a more mature we need to look into having defaults vs throwing errors. Balancing between having verbose code and ensuring stuff does not "fail silently"